x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAlgorithmValidator.java

```diff
@@ -28,15 +28,16 @@ public JwtAlgorithmValidator(List<String> allowedAlgorithms) {
     public void validate(JWSHeader jwsHeader, JWTClaimsSet jwtClaimsSet) {
         final JWSAlgorithm algorithm = jwsHeader.getAlgorithm();
         if (algorithm == null) {
-            throw new ElasticsearchSecurityException("missing JWT algorithm header", RestStatus.BAD_REQUEST);
+            throw new IllegalArgumentException("missing JWT algorithm header");
```

Comment:
I replaced `ElasticsearchSecurityException` with `IllegalArgumentException` here and in quite a few other places to simplify things. Validation errors are printed in logs. In that context, showing the direct underlying cause (illegal argument) is more appropriate. For user response, since this fails authentication, it is always a 401 `security_exception`.

Reply:
No it's only used in logs. See 
https://github.com/elastic/elasticsearch/blob/a334a530efa1460872ebb2bf53706cba3fbdac9e/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RealmsAuthenticator.java#L189
and
https://github.com/elastic/elasticsearch/blob/a334a530efa1460872ebb2bf53706cba3fbdac9e/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RealmsAuthenticator.java#L256-L260

When authentication fails, it's intentional to send little information back to the caller for security reasons. The detail exception is only available in server logs.